src/snowflake/snowpark/types.py

```diff
@@ -41,6 +42,23 @@ def __repr__(self):
     def is_primitive(self):
         return True
 
+    @classmethod
+    def type_name(cls) -> str:
+        return cls.__name__[:-4].lower()
+
+    def simple_string(self) -> str:
+        return self.type_name()
+
+    def json_value(self) -> Union[str, Dict[str, Any]]:
+        return self.type_name()
+
+    def json(self) -> str:
+        return json.dumps(self.json_value(), separators=(",", ":"), sort_keys=True)
+
+    typeName = type_name
+    simpleString = simple_string
+    jsonValue = json_value
```

Comment:
can we implement these using classmethod as well so we don't have to re-add aliases each time.

Reply:
I think there two reasons we cannot do that, correct me if I was wrong.
1. in pyspark, only type_name is classmethod, do we want this different behavior?
2. even if we make them classmethod, we still need to re-add aliases each time because the aliases will always point to functions in DataType(parent class) if we don't re-add them.

Reply:
I think you can use something like this to call the class version of `simple_string`

https://github.com/snowflakedb/snowpark-python/blob/8b6090ab2e2fde1408afa38693b3d223970890d7/src/snowflake/snowpark/mock/_functions.py#L140-L145

Reply:
discussed offline, we decide to keep what we have now since these function need variable from instance and cannot be made classmethod